### PR TITLE
Bluetooth: crypto_toolbox: Move out Crypto Toolbox functions of host

### PR DESCRIPTION
**Pull request by theob-pro**
Create crypto_toolbox library that contain Crypto Toolbox functions from the BT specification.

Status:
Code is ready for review.
I will clean up the git history later.

### CODE REVIEW

File: subsys/bluetooth/crypto_toolbox/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+zephyr_library_sources_ifdef(CONFIG_BT_CRYPTO_TOOLBOX
+  crypto_toolbox_aes_cmac.c
+  crypto_toolbox_f4.c
+  crypto_toolbox_f5.c
+  crypto_toolbox_f6.c
+  crypto_toolbox_g2.c
+)
+
+if(CONFIG_BT_BREDR)
```
Comment (alwa-nordic) [inline subsys/bluetooth/crypto_toolbox/CMakeLists.txt:18]: I would prefer if crypto toolbox does not reference CONFIG_BT_BREDR. Let's discuss options offline.

File: subsys/bluetooth/crypto_toolbox/Kconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
+config BT_CRYPTO_TOOLBOX
+    bool "Cryptographic Toolbox"
```
Comment (alwa-nordic) [inline subsys/bluetooth/crypto_toolbox/Kconfig:2]: I think this should not have a prompt. I'm my mind this is not a user option.
```suggestion
    bool
```

File: subsys/bluetooth/crypto_toolbox/Kconfig [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    select TINYCRYPT
+    select TINYCRYPT_AES
+    select TINYCRYPT_AES_CMAC
+    help
+      This option enable the Bluetooth Cryptographic Toolbox.
```
Comment (alwa-nordic) [inline subsys/bluetooth/crypto_toolbox/Kconfig:7]:
```suggestion
      This option enables the Bluetooth Cryptographic Toolbox.
```

File: subsys/bluetooth/crypto_toolbox/Kconfig [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+      
+config BT_DEBUG_CRYPTO_TOOLBOX
+	bool "Bluetooth Cryptographic Toolbox debug"
+	depends on BT_CRYPTO_TOOLBOX
+	help
+	  This option enables debug support for the Bluetooth
```
Comment (alwa-nordic) [inline subsys/bluetooth/crypto_toolbox/Kconfig:13]:
```suggestion
	  This option enables debug log output for the Bluetooth
```

File: subsys/bluetooth/crypto_toolbox/crypto_toolbox_f4.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/bluetooth/crypto_toolbox/f4.h>
+
+#include <zephyr/bluetooth/crypto_toolbox/aes_cmac.h>
```
Comment (alwa-nordic) [inline subsys/bluetooth/crypto_toolbox/crypto_toolbox_f4.c:4]: Please run clang-format on all the new files.

File: subsys/bluetooth/crypto_toolbox/include/zephyr/bluetooth/crypto_toolbox/h6.h [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * @brief Cryptographic Toolbox h6
+ *
+ * Link key conversion defined in Core Vol. 3, part H 2.2.10.
+ *
+ * @param w 128-bit key
+ * @param key_id 32-bit
+ * @param[out] res 128-bit
+ * @retval 0 Computation was successful. @p res contains the result.
+ * @retval @-EIO
```
Comment (alwa-nordic) [inline subsys/bluetooth/crypto_toolbox/include/zephyr/bluetooth/crypto_toolbox/h6.h:12]: The `@` should not be there.

**Review by alwa-nordic — CHANGES_REQUESTED** [review]

**Review by carlescufi — CHANGES_REQUESTED** [review]
Why `crypto_toolbox`? It is unnecessary long?
I would suggest `bt_crypto` or `host_crypto`.

**Comment by carlescufi** [comment]
Superseded by #51809

**Pull request closed without merging**